ouroboros-consensus/src/Ouroboros/Consensus/Mempool/TxSeq.hs

```diff
@@ -144,3 +148,17 @@ splitAfterTicketNo (TxSeq txs) n =
     case FingerTree.split (\m -> mMaxTicket m > n) txs of
       (l, r) -> (TxSeq l, TxSeq r)
 
+-- | Convert a 'TxSeq' to the type which is expected to be returned from the
+-- 'Mempool' API.
+fromTxSeq :: TxSeq tx -> Seq (tx, TicketNo)
+fromTxSeq (TxSeq ftree) = fmap
+  (\(TxTicket tx tn) -> (tx, tn))
+  (Seq.fromList . Foldable.toList $ ftree)
+
+-- | Append a @tx@ to the back of a 'TxSeq'.
+-- n.b. This function also handles the incrementing of the newly added
+-- transaction's ticket number.
+appendTx :: TxSeq tx -> tx -> TxSeq tx
```

Comment:
Ah wait, this isn't going to work. The mempool impl will have to maintain the `TicketNo` counter separately. The mempool can become empty during operation, but I think we do not want to reset the `TicketNo` to 0 (or rather 1) when it is empty. Or at least I don't have the brain power to think through whether the mempool readers would correctly deal with the counter resetting.